File: Arashyn-BE/src/main/java/com/arashi/edu/arashynbe/features/auth/service/AuthService.java
package com.arashi.edu.arashynbe.features.auth.service;

import com.arashi.edu.arashynbe.features.auth.dto.AuthResult;
import com.arashi.edu.arashynbe.features.auth.dto.request.LoginRequest;
import com.arashi.edu.arashynbe.features.auth.dto.request.RegisterRequest;
import com.arashi.edu.arashynbe.features.auth.dto.response.RegisterResponse;

public interface AuthService {

  RegisterResponse register(RegisterRequest request);

  AuthResult login(LoginRequest request);

  AuthResult refresh(String refreshToken);

  void logout(String refreshToken);
}

File: Arashyn-FE/src/entities/auth/auth_service.ts
import { api } from "@/lib/api/request.ts";
import type {
    LoginResponse,
    RegisterResponse,
    RegisterVerifyPayload,
    VerifyOtpPayload,
    ResendVerificationPayload,
    CompleteRegisterPayload,
    LoginPayload,
} from "./auth_types.ts";

export const authService = {
    verifyRegister: (payload: RegisterVerifyPayload) =>
        api.post<void>("/auth/verify", payload),

    /** POST /auth/verify/confirm */
    confirmVerify: (payload: VerifyOtpPayload) =>
        api.post<void>("/auth/verify/confirm", payload),

    /** POST /auth/verify/resend */
    resendVerification: (payload: ResendVerificationPayload) =>
        api.post<void>("/auth/verify/resend", payload),

    /** POST /auth/complete-register */
    completeRegister: (payload: CompleteRegisterPayload) =>
        api.post<RegisterResponse>("/auth/complete-register", payload),

    /** POST /auth/login */
    login: (payload: LoginPayload) =>
        api.post<LoginResponse>("/auth/login", payload),

    /** POST /auth/refresh */
    refresh: () =>
        api.post<LoginResponse>("/auth/refresh"),

    /** POST /auth/logout */
    logout: () =>
        api.post<void>("/auth/logout"),
};